python/pyspark/sql/connect/dataframe.py

```diff
@@ -201,6 +202,34 @@ def sort(self, *cols: "ColumnOrString") -> "DataFrame":
         """Sort by a specific column"""
         return DataFrame.withPlan(plan.Sort(self._plan, *cols), session=self._session)
 
+    def sample(
+        self,
+        withReplacement: Optional[Union[float, bool]] = None,
+        fraction: Optional[Union[int, float]] = None,
+        seed: Optional[int] = None,
+    ) -> "DataFrame":
+        from pyspark.sql import DataFrame as PySparkDataFrame
```

Comment:
oh, does spark connect python client depends on pyspark? Then it's not a thin client any more...

Reply:
yes this is now depending on the pyspark. In fact since the first PR.  For the short term it is ok cc @HyukjinKwon 

I guess we will need to make a final decision how whether it should depend or not before making the python packaging and release.